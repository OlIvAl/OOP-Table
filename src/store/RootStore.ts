import { FinanceActionsStore } from './Actions/FinanceActionsStore';
import { IFinanceActionModel } from './Actions/interfaces';
import { IStoreWithCurrentModel } from './Abstract/interfaces';

export class RootStore {
  financeActionsStore: IStoreWithCurrentModel<IFinanceActionModel>;

  constructor() {
    this.financeActionsStore = new FinanceActionsStore(this);
  }
}