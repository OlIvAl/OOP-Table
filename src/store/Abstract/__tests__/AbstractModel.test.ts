import TestStore from '../Test/TestStore';
import { IStore } from '../interfaces';
import { ITestModel } from '../Test/interfaces';
import TestModel from '../Test/TestModel';
import { RootStore } from '../../RootStore';

let rootStore = new RootStore();
let store: IStore<ITestModel>;
let model: ITestModel;

beforeEach(() => {
  store = new TestStore(rootStore);
  model = new TestModel(store);
});

describe('AbstractModel', () => {
  describe('constructor', () => {
    it('init model with valid store', () => {
      expect(model).toMatchSnapshot();
      expect(model.store).toEqual(store);
    });
  });
});
