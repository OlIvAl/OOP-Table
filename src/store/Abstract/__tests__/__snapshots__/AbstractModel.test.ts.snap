// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractModel constructor init model with valid store 1`] = `
TestModel {
  "store": TestStore {
    "rootStore": RootStore {
      "financeActionsStore": FinanceActionsStore {
        "rootStore": [Circular],
      },
    },
  },
}
`;
